changeset.go

```diff
@@ -1193,6 +1331,58 @@ func (cs *Changeset) revertClusterRole(ctx context.Context, item *ChangesetItem)
 	return control.Upsert(ctx)
 }
 
+func (cs *Changeset) revertNamespace(ctx context.Context, item *ChangesetItem) error {
+	resource := item.From
+	if len(resource) == 0 {
+		resource = item.To
+	}
+	namespace, err := ParseNamespace(strings.NewReader(resource))
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	control, err := NewNamespaceControl(NamespaceConfig{Namespace: namespace, Client: cs.Client})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	// this operation created the resource, so we will delete it
+	if len(item.From) == 0 {
+		err = control.Delete(ctx, true)
+		// If the resource has already been deleted, suppress the error
+		if trace.IsNotFound(err) {
+			return nil
+		}
+		return trace.Wrap(err)
+	}
+	// this operation either created or updated the resource, so we create a new version
```

Comment:
Nit: I wonder if the comment should actually say "either **deleted** or updated" b/c the "created" case is handled above.